izumi-reflect/izumi-reflect/src/main/scala-3/izumi/reflect/TagMacro.scala

```diff
@@ -64,27 +62,42 @@ final class TagMacro(using override val qctx: Quotes) extends InspectorBase {
         val ltts: Expr[List[LightTypeTag]] = Expr.ofList(ltts0)
         '{ Tag.unionTag[T](classOf[Any], $ltts) }
 
+      case x if x.typeSymbol.isTypeParam => summonTag(x)
+
       case _ =>
         throw new Exception(s"Unsupported type: $typeRepr")
     }
 
   private def summonTag[any <: AnyKind](typeRepr: TypeRepr): Expr[Tag[any]] =
+    summonTagEither(typeRepr) match {
+      case Right(tag) => tag.asInstanceOf[Expr[Tag[any]]]
+      case Left(message) => report.errorAndAbort(message)
+    }
+
+  private def summonTagEither[any <: AnyKind](typeRepr: TypeRepr): Either[String, Expr[Tag[any]]] =
     typeRepr.asType match {
       case given Type[a] =>
         val message = s"Cannot find implicit Tag[${Type.show[a]}]"
-        Expr.summon[Tag[a]].getOrElse(report.errorAndAbort(message)).asInstanceOf[Expr[Tag[any]]]
+        Expr.summon[Tag[a]] match {
+          case Some(tag) => Right(tag.asInstanceOf[Expr[Tag[any]]])
+          case None => Left(message)
+        }
     }
 
+  private def summonable[any <: AnyKind](typeRepr: TypeRepr): Boolean =
+    summonTagEither(typeRepr).isRight
+
   /**
     * Returns true if the given type contains no type parameters
     * (this means the type is not "weak" https://stackoverflow.com/questions/29435985/weaktypetag-v-typetag)
     */
   private def allPartsStrong(typeRepr: TypeRepr): Boolean =
     typeRepr.dealias match {
-      case x if x.typeSymbol.isTypeParam => false
+      case x if x.typeSymbol.isTypeParam && summonable(x) => false
```

Comment:
Is there a way to avoid implicit search here? In what situation do we need to regard a type parameter as strong if it doesn't have an implicit Tag?

Reply:
I was trying to fix this and finally it had worked without implicit search 🎉 after merging #279.